### ISSUE TITLE
Unbundle specifications from versioned releases

### ISSUE THREAD

**Issue by attilamolnar** [ircv3/ircv3-specifications#265]
Specifications should be released when they are ready. There is little point in bundling a bunch of unrelated specifications together, and it is also slowing down the ecosystem because finished minor specifications are being held back because work is underway on an unrelated major specification.

Related: https://github.com/ircv3/ircv3-specifications/issues/221#issuecomment-183365232

**Comment by jwheare** [comment]
My thoughts:

This should apply from the 3.3 batch of specifications going forward. i.e. we should drop the use of identifiers like 3.x entirely.

The only purpose this served was to indicate that the previous version in some way was a prerequisite for support. Specifications instead should list their dependencies explicitly. This also removes the need for the Base vs Optional distinction. If a "base" specification is needed, it can be listed as such, rather than implied by a version number.

Some examples:
* STS depends on Capability negotiation 3.2, CAP LS 302 (due to CAP values)
* The `message-tags` CAP depends on the 3.2 Message Tags spec.

In the case of `message-tags`, I think it's also worth including the parts of the previous spec that still apply, and mark the previous version as deprecated/obsolete, to avoid implementors having to consult a chain of specs in order to understand the latest version.

This shouldn't apply where a spec merely uses a mechanism specified elsewhere (e.g. specs don't all need to include the details of CAP negotiation)

In summary:

* Drop version numbers
* Drop optional/base distinction
* List dependencies explicitly
* Release specifications independently when they're ratified

**Comment by jwheare** [comment]
This is replaced by #329
